### PR TITLE
Initially hiding non-checked filters

### PR DESCRIPTION
**Pull request by AdamZarger**
Added a SimpleFilter component for checkbox filters. Added logic to expand/collapse filters based on URL parameters on initial page load.

## Additions

- SimpleFilter and SimpleCollapsibleFilter React components
- SimpleFilters are automatically collapsed and will automatically expand if a particular filter is present in the URL parameters

## Testing

- To Do

## Review

- @JeffreyMFarley 
- @amymok 

## Todos

- Fix broken tests
- Write tests for new components

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

### CODE REVIEW

File: src/Filters/SimpleCollapsibleFilter.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+export default class SimpleCollapsibleFilter extends React.Component {
```
Comment (JeffreyMFarley) [inline src/Filters/SimpleCollapsibleFilter.jsx]: I don't think you need the additional class.  You should be able to fix `CollapsibleFilter`

File: src/Filters/SimpleCollapsibleFilter.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      showChildren: typeof props.showChildren === 'undefined' ? false : props.showChildren
```
Comment (JeffreyMFarley) [inline src/Filters/SimpleCollapsibleFilter.jsx]: This line should be `showChildren: props.showChildren`

At the bottom, there should be something like:

```Javascript
CollapsibleFilter.propTypes = {
  showChildren: PropTypes.bool
}

CollapsibleFilter.defaultProps = {
  showChildren: true
}
```

This way, the default behavior is 'shown' but can be overridden, like is done in `SimpleFilter`

File: src/Filters/SimpleFilter.jsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  }
+}
+
+export const mapStateToProps = (state, ownProps) => {
+  // Find all query filters that refer to the field name
+  const activeChildren = state.query[ownProps.fieldName] || []
+
+  return {
+    options: state.aggs[ownProps.fieldName] || [],
+    showChildren: activeChildren.length > 0
```
Comment (JeffreyMFarley) [inline src/Filters/SimpleFilter.jsx]: 👍

**Comment by JeffreyMFarley** [comment]
Awesome! Now just get back to 100% coverage and we are good to go

**Comment by AdamZarger** [comment]
My mistake, I didn't realize I had pushed these changes. Thanks for your suggestions! As you saw I was able to implement them without needing the custom filter. I'm working on adapting the UrlBar tests now

File: src/Filters/__tests__/SimpleFilter.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  });
+
+  describe('componentWillReceiveProps', () => {
+    it('changes props correctly', () => {
+      props.params.timely = 'no'
+      const expected = 'no'
+
+      target.componentWillReceiveProps(props)
+
+      expect(target.props.params.timely).toEqual(expected)
```
Comment (JeffreyMFarley) [inline src/Filters/__tests__/SimpleFilter.spec.jsx]: I realize you borrowed this code from `UrlBarSynch`, but it won't work here, and that's why you get this error:

```
 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the SimpleFilter component.
```
This is more like how the test should work:

```javascript
      const {target} = setupEnzyme()
      target.setProps({showChildren: true})
      expect(target.state('showChildren')).toEqual(true)

```

What is `setupEnzyme`? There are a lot of examples in the codebase...

Comment (AdamZarger) [inline src/Filters/__tests__/SimpleFilter.spec.jsx]: Thank you for this feedback! I was pretty sure that there was a better way than how I was doing this. I was able to use a different example of `setupEnzyme` to figure out how to properly write this test

**Pull request merged**